new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -0,0 +1,138 @@
+import {DecodeOptionsType} from "../typings/decoders";
+import {
+    CursorType,
+    FileSrcType,
+    LineNumLogEventNumMap,
+} from "../typings/worker";
+import {getUint8ArrayFrom} from "../utils/http";
+import {getBasenameFromUrl} from "../utils/url";
+import {
+    Decoder,
+    DecoderConstructor,
+} from "./decoders/Decoder";
+import JsonlDecoder from "./decoders/JsonlDecoder";
+
+
+/**
+ * A mapping of file name extensions to their respective decoders.
+ */
+const FILE_EXT_TO_DECODER: Readonly<Record<string, DecoderConstructor>> = Object.freeze({
+    ".jsonl": JsonlDecoder as unknown as DecoderConstructor,
+});
+
+class LogFileManager {
+    #pageSize: number;
+
+    #fileData: Uint8Array | null = null;
+
+    #fileName: string | null = null;
+
+    #numEvents: number = 0;
+
+    #decoder: Decoder | null = null;
+
+
+    /**
+     * Retrieves the decoder constructor based on the file extension.
+     *
+     * @param filename The filename from which to extract the file extension.
+     * @return The constructor associated with the file extension, or null if not found.
+     */
+    static #getDecoderConstructor = (filename: string): DecoderConstructor | null => {
+        for (const ext in FILE_EXT_TO_DECODER) {
+            if (Object.hasOwn(FILE_EXT_TO_DECODER, ext)) {
+                const decoderConstructor = FILE_EXT_TO_DECODER[ext];
+                if (filename.endsWith(ext) && "undefined" !== typeof decoderConstructor) {
+                    return decoderConstructor;
+                }
+            }
+        }
+
+        return null;
+    };
+
+    constructor (pageSize: number) {
+        this.#pageSize = pageSize;
+    }
+
+    async loadFile (fileSrc: FileSrcType): Promise<number> {
```

Comment:
```suggestion
    /**
    * Loads a file from a given source and decodes it using the appropriate decoder based on the file extension.
    *
    * @param {FileSrcType} fileSrc - The source of the file to load. This can be a string representing a URL, or a File object.
    * @return {Promise<number>} - A promise that resolves with the number of log events found in the file.
    * @throws {Error} - Throws an error if the file source type is not supported.
    */
    async loadFile (fileSrc: FileSrcType): Promise<number> {
```

Reply:
> @throws {Error} - Throws an error if the file source type is not supported.

I guess we need to change Line 64 to throw instead, right?